### DISCUSSION TITLE
Support CSS pseudo-elements (e.g. `::before`) in `RemoteWebElement::getText()`

### DISCUSSION THREAD

**Discussion by ThomasLandauer** [php-webdriver/php-webdriver#901] [Ideas]
When this is the HTML:
```html
<!DOCTYPE html>
<html>
    <head>
        <style>
            .c1::before { content:"f"; }
        </style>
    </head>
    <body>
        <span class="c1">oo</span>
    </body>
</html>
```
... the output of [Symfony Panther](https://github.com/symfony/panther)
```php
$crawler = $client->request('GET', '...');
dump($crawler->text());
```
is:
```
oo
```
... while the rendered text in the browser is:
```
foo
```

The specs at https://www.w3.org/TR/webdriver/#get-element-text are saying:

> The Get Element Text command intends to return an element’s text “as rendered”.

... so I would say it should be `foo`. If that's not possible due to backwards compatibility with Selenium's [`bot.dom.getVisibleText`](https://github.com/SeleniumHQ/selenium/blob/e09e28f016c9f53196cf68d6f71991c5af4a35d4/javascript/atoms/dom.js#L981), then maybe create a new method for that?

**Comment by OndraM** [comment]
Hi, this seems as a behavior of the browser drivers. I tried both Chromedriver and Geckodriver, and both of them do read the element text as "oo". Ie. they don't prepend the "f" from css pseudo-element. 

Even though the specification may sound it should return the value including CSS modifications, I'm not quite sure whether this is not intentional behavior.  Anyway, the current behavior is based on the value the browser driver returns for the "get text" webdriver command. We don't manipulate the returned value in any way, so I'm not sure this should be handled by php-webdriver. I'm not even sure there is any other way except some javascript to read the value from ::before or ::after.

**Reply by ThomasLandauer** [reply]
Thanks! I opened an issue at https://github.com/mozilla/geckodriver/issues/1879 - but I've never used geckodriver "directly", so I hope they don't have any questions ;-)
